File: src/main/java/me/testaccount666/migration/essentials/WarpMigrator.java
package me.testaccount666.migration.essentials;

import com.earth2me.essentials.commands.WarpNotFoundException;
import me.testaccount666.serversystem.ServerSystem;
import me.testaccount666.serversystem.commands.executables.warp.manager.WarpManager;

import java.util.logging.Level;

public class WarpMigrator extends AbstractMigrator {

    @Override
    public int migrateFrom() {
        var warpManager = ServerSystem.Instance.getRegistry().getService(WarpManager.class);
        var essentials = essentials();

        var count = 0;
        for (var warpName : essentials.getWarps().getList())
            try {
                var location = essentials.getWarps().getWarp(warpName);
                warpManager.addWarp(warpName, location);

                count += 1;
            } catch (IllegalArgumentException | WarpNotFoundException exception) {
                ServerSystem.getLog().log(Level.WARNING, "Couldn't migrate warp '${warpName}'", exception);
            }

        return count;
    }

    @Override
    public int migrateTo() {
        var warpManager = ServerSystem.Instance.getRegistry().getService(WarpManager.class);
        var essentials = essentials();

        var count = 0;
        for (var warp : warpManager.getWarps())
            try {
                var warpName = warp.getDisplayName();
                var location = warp.getLocation();

                essentials.getWarps().setWarp(warpName, location);

                count += 1;
            } catch (Exception exception) {
                ServerSystem.getLog().log(Level.WARNING, "Couldn't migrate warp '${warp.getDisplayName()}'", exception);
            }

        return count;
    }
}

File: src/main/kotlin/me/testaccount666/migration/plugins/essentials/WarpMigrator.kt
package me.testaccount666.migration.plugins.essentials

import me.testaccount666.serversystem.ServerSystem.Companion.instance
import me.testaccount666.serversystem.ServerSystem.Companion.log
import me.testaccount666.serversystem.commands.executables.waypoints.warp.manager.WarpManager
import java.util.logging.Level

class WarpMigrator : AbstractMigrator() {
    val warpManager by lazy { instance.registry.getService<WarpManager>() }
    override fun migrateFrom(): Int {
        val essentials = essentials

        val count = essentials.warps.list.count { warpName ->
            runCatching {
                val location = essentials.warps.getWarp(warpName)
                warpManager.addPoint(warpName, location)

                return@runCatching true
            }.onFailure { log.log(Level.WARNING, "Couldn't migrate warp '$warpName'", it) }.getOrDefault(false)
        }

        return count
    }

    override fun migrateTo(): Int {
        val essentials = essentials

        val count = warpManager.waypoints.count { warp ->
            runCatching {
                val warpName = warp.displayName
                val location = warp.location

                essentials.warps.setWarp(warpName, location)

                return@runCatching true
            }.onFailure { log.log(Level.WARNING, "Couldn't migrate warp '${warp.displayName}'", it) }.getOrDefault(false)
        }

        return count
    }
}
